server/cmwell-ws/app/controllers/Application.scala

```diff
@@ -3004,6 +3001,27 @@ callback=< [URL] >
     }
   }
 
+  def handleKafkaConsume(topic: String, partition: Int): Action[AnyContent] = Action { implicit req =>
+    val sysTopics = Set("persist_topic", "index_topic", "persist_topic.priority", "index_topic.priority")
+    if (sysTopics(topic) && !authUtils.isOperationAllowedForUser(Admin, authUtils.extractTokenFrom(req), evenForNonProdEnv = true))
+      Forbidden("Consuming this topic requires an Admin token.")
+    else {
+      val offset = req.getQueryString("offset").fold(0L)(_.toLong)
+      val maxLengthOpt = req.getQueryString("max-length").map(_.toLong)
+      val isText = req.getQueryString("format").fold(false)(_ == "text")
+
+      val source = crudServiceFS.consumeKafka(topic, partition, offset, maxLengthOpt)
+      Ok.chunked(source.map { bytes =>
+        (Try(CommandSerializer.decode(bytes)) match {
```

Comment:
For the Crawler topic it will fail (throw an exception???) for each message and only then will return the actual message.

Also for the crawler topic it will get into:
https://github.com/thomsonreuters/CM-Well/blob/04039b236a43dc2e86990a80d7cf97e0946b545d/server/cmwell-common/src/main/scala/cmwell/common/formats/JsonSerializer.scala#L133
And will throw an exception in the server.

Reply:
won't it get to that line: `case Failure(_) if isText => new String(bytes, StandardCharsets.UTF_8)` ?

Reply:
might be but it will throw an exception for **each** element!